frontend/src/App.jsx

```diff
@@ -1,19 +1,74 @@
-import { BrowserRouter, Routes, Route } from "react-router-dom";
+import { BrowserRouter, Routes, Route, Navigate } from "react-router-dom";
 import "./App.css";
 import Login from "./Login";
 import SignUp from "./SignUp";
 import Home from "./Home";
-import { useState } from "react";
+import { useState, useEffect } from "react";
+import { BASE_URL } from "./utils/reused";
 
 const App = () => {
+  const [isLoggedIn, setLoggedIn] = useState(false);
+
+  const attemptLogin = async () => {
+    try {
+      const response = await fetch(`${BASE_URL}/auth/me`, {
+        method: "GET",
+        credentials: "include",
+      });
+      if (response.ok) {
+        setLoggedIn(true);
+      }
+    } catch {
+      return;
+    }
+  };
+
+  attemptLogin();
```

Comment:
Why do you run this on every render? Here it's probably a good idea to add `useEffect`, with a dependency on `isLoggedIn`.